### PR TITLE
adding support for looksrare sales

### PR DESCRIPTION
**Pull request by Theo0x1337**
Adding support for LooksRare sales, including the following changes:

- Refactoring variables name (e.g: openSeaSales to sales)
- Refactoring files name
- Updating types to match the entities of the new subgraph indexing LooksRare sales
- Adding an option to filter on exchange (OS_V1, OS_V2, OS or LR_V1)
- Adding LooksRare volume to CSV headers

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201932664610968

### CODE REVIEW

**Comment by jakerockland** [comment]
Sending this your way @grantoesterling @Asupkay to review for when y'all have bandwidth 🙏

File: RoyaltiesProcessorDelivery/src/repositories/opensea_api.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ function openSeaEventModelToSubgraphModel(
       }
     }
     // Convert token(s) to array of subgraph's T_OpenSeaSaleLookupTable model
```
Comment (grantoesterling) [inline RoyaltiesProcessorDelivery/src/repositories/opensea_api.ts:101]:
```suggestion
    // Convert token(s) to array of subgraph's T_SaleLookupTable model

```

**Review by grantoesterling — APPROVED** [review]

**Review by ryley-o — APPROVED** [review]
TYVM!
Just a few nits and couple minor suggestions, mostly around some of the weirdness associated with OpenSea API-mode.

Obviously this is not tested due to the LooksRare subgraph not yet being deployed, but I think main branch is already ahead of our subgraph deployment, so okay to merge. New subgraph deployment will happen soon.

File: RoyaltiesProcessorDelivery/src/index.ts [hunk 2]
```diff
@@ -461,15 +481,15 @@ yargs(hideBin(process.argv))
       }
```
[the remark this answers is not in the document]
Comment (Theo0x1337) [reply RoyaltiesProcessorDelivery/src/index.ts:483]: fixed here @[14fe50c](https://github.com/ArtBlocks/artblocks-community-tooling/pull/20/commits/14fe50c1ef75d4632d5183b899b2dd618524fc66)

File: RoyaltiesProcessorDelivery/src/types/project_report.ts [hunk 3]
```diff
@@ -47,34 +47,35 @@ export class ProjectReport {
...
+        const cryptoName = addressToPaymentToken(paymentToken);
+
+        let volume = this.#paymentTokenVolumes.get(cryptoName);
+        if (volume === undefined) {
+            volume = {
+                total: BigNumber.from(0),
+                OS_V1: BigNumber.from(0),
+                OS_V2: BigNumber.from(0),
+                LR_V1: BigNumber.from(0),
+                Vunknown: BigNumber.from(0), // when using OS API, version is unknown
```
[the remark this answers is not in the document]
Comment (Theo0x1337) [reply RoyaltiesProcessorDelivery/src/types/project_report.ts]: Done here @[f13c4fc](https://github.com/ArtBlocks/artblocks-community-tooling/pull/20/commits/f13c4fcd149fecb24fdbd8ebe1a1b2f9062cf861)

File: RoyaltiesProcessorDelivery/src/repositories/opensea_api.ts [hunk 4]
```diff
@@ -213,27 +212,27 @@ function openSeaEventModelToSubgraphModel(
...
-    return _sale;
-  } catch (e) {
-    console.error("[ERROR] Error in OpenSeaSaleConverter");
-    console.error(_event)
-    console.error(_event.payment_token)
-    throw(e)
-  }
+      const _sale: T_Sale = {
+        id: _event.id,
+        exchange: "Vunknown",
```
[the remark this answers is not in the document]
Comment (Theo0x1337) [reply RoyaltiesProcessorDelivery/src/repositories/opensea_api.ts]: Done here @[f13c4fc](https://github.com/ArtBlocks/artblocks-community-tooling/pull/20/commits/f13c4fcd149fecb24fdbd8ebe1a1b2f9062cf861)

File: RoyaltiesProcessorDelivery/src/index.ts [hunk 5]
```diff
@@ -123,50 +125,54 @@ async function processSales(
...
   }
 
-  console.info(`[INFO] ${openSeaSales.length} OpenSea sales have been fetched`);
+  console.info(`[INFO] ${sales.length} sales have been fetched`);
 
   // filter out all subgraph bundles with > 1 OpenSea collection slug
   // (in series to slow down because OpenSea API is rate limited)
   console.info(
-    "[INFO] Getting OpenSea collection slugs for tokens in bundle sales... (cached upon receipt)"
+    "[INFO] Getting collection slugs for tokens in bundle sales... (cached upon receipt)"
```
[the remark this answers is not in the document]
Comment (Theo0x1337) [reply RoyaltiesProcessorDelivery/src/index.ts]: Done here @[cfd42af](https://github.com/ArtBlocks/artblocks-community-tooling/pull/20/commits/cfd42affd08d807e384d75108356fd1958dc12d6)

**Comment by Theo0x1337** [comment]
Fixed all above comments as you suggested! Also changed the readme file to fit the new `exchange` command and support for LooksRare.

**Comment by ryley-o** [comment]
@Theo0x1337 LGTM! Thank you for implementing this!

**Comment by ryley-o** [comment]
(just to note - I did regression test this against our May reports - everything matched up perfectly when testing in OpenSea API mode!)

**Pull request merged**